From: Takashi Iwai <tiwai@suse.de>
Subject: dump-module: Ignore invalid module-init-tools entries
References: bsc#1194379

pulseaudio dump-modules iterates all entries beginning with "module-"
prefix, and unfortunately the directory /usr/lib/module-init-tools
that is installed by suse-module-tools package hitting incorrectly,
which results in spurious errors as:
  E: [pulseaudio] ltdl-bind-now.c: Failed to open module /lib/module-init-tools.so: /lib/module-init-tools.so: cannot open shared object file: No such file or directory
  ....

Simply ignore this entry as a workaround.

Signed-off-by: Takashi Iwai <tiwai@suse.de>

---
diff --git a/src/daemon/dumpmodules.c b/src/daemon/dumpmodules.c
index 8410bbcdcca8..2abc3b8bb0c9 100644
--- a/src/daemon/dumpmodules.c
+++ b/src/daemon/dumpmodules.c
@@ -115,6 +115,8 @@ static int callback(const char *path, lt_ptr data) {
 
     if (strlen(e) <= sizeof(PREFIX)-1 || strncmp(e, PREFIX, sizeof(PREFIX)-1))
         return 0;
+    if (!strcmp(e, "module-init-tools"))
+        return 0;
 
     if (is_preloaded(e))
         return 0;
@@ -141,6 +143,8 @@ void pa_dump_modules(pa_daemon_conf *c, int argc, char * const argv[]) {
 
             if (strlen(l->name) <= sizeof(PREFIX)-1 || strncmp(l->name, PREFIX, sizeof(PREFIX)-1))
                 continue;
+            if (!strcmp(l->name, "module-init-tools"))
+                continue;
 
             pa_snprintf(buf, sizeof(buf), "%s", l->name);
             if ((e = strrchr(buf, '.')))
